### PR TITLE
Add `content_object` index to `Link`

### LINKED ISSUE #202: Add index to `Link`
The [django documentation](https://docs.djangoproject.com/en/5.1/ref/contrib/contenttypes/#django.contrib.contenttypes.fields.GenericForeignKey) suggests to always create an index on the `content_type` and `object_id`  fields.
However, currently, there is no index for the `Link` model: https://github.com/DjangoAdminHackers/django-linkcheck/blob/d29f55b3f5fd790f4183bb2fe5eb67055f856f7e/linkcheck/models.py#L513 
At https://github.com/digitalfabrik/integreat-cms, a slow query using `Link` objects can be speed up about 40% by adding this index.

## Solution

Add this index to the `Link` model:

```python
    indexes = [
            models.Index(fields=["content_type", "object_id"]),
        ]
```

### PR DESCRIPTION
**Pull request by david-venhoff**
This pr adds an index to the `Link` model for faster queries.

Fixes: #202

### CODE REVIEW

File: linkcheck/models.py [hunk 1]
```diff
@@ -542,6 +542,11 @@ def __str__(self):
     def __repr__(self):
         return f"<Link (id: {self.id}, url: {self.url!r}, source: {self.content_object!r})>"
 
+    class Meta:
+        indexes = [
+            models.Index(fields=["content_type", "object_id"]),
```
Comment (claudep) [inline linkcheck/models.py]: I have a slight preference to adding an explicit name to the index, what do you think?

And generally I place the Meta class just after the field definitions. Is there a good practice to put it like you did?

Comment (david-venhoff) [reply linkcheck/models.py]: +1 one the name for the index.

Regarding the meta class position, I am not aware of style guidelines specific to that. It's just how we do it at at https://github.com/digitalfabrik/integreat-cms and I did not really think about it :smile:

**Comment by claudep** [comment]
Sorry for the test failure, it's a bug in `.github/workflows/test.yml`. I'll work on a fix ASAP.

**Comment by claudep** [comment]
OK, done. Could you please rebase your patch so we can test again your code?

**Comment by david-venhoff** [comment]
Done

**Review by claudep — APPROVED** [review]
Thanks !!

**Pull request merged**